src/AutoRest.CSharp/Common/Input/CodeModelPartials.cs

```diff
@@ -276,11 +276,20 @@ internal partial class CodeModel
         [YamlDotNet.Serialization.YamlMember(Alias = "testModel")]
         public TestModel? TestModel { get; set; }
 
-        private IEnumerable<Schema>? _allSchemas;
-        public IEnumerable<Schema> AllSchemas => _allSchemas ??= Schemas.Choices.Cast<Schema>()
-                .Concat(Schemas.SealedChoices)
-                .Concat(Schemas.Objects)
-                .Concat(Schemas.Groups);
+        public IEnumerable<Schema> AllSchemas
+        {
+            get
+            {
+                foreach (var schema in Schemas.Choices)
+                    yield return schema;
+                foreach (var schema in Schemas.SealedChoices)
+                    yield return schema;
+                foreach (var schema in Schemas.Objects)
+                    yield return schema;
+                foreach (var schema in Schemas.Groups)
+                    yield return schema;
+            }
```

Comment:
does it need explicitly return null?

Reply:
This is an equivalent version of "return a collection with all the 4 piece concatenated together". This will never return a null value because we have explicit null checks